app/javascript/dashboard/components/NetworkNotification.vue

```diff
@@ -47,7 +57,12 @@ export default {
   mounted() {
     window.addEventListener('offline', this.updateOnlineStatus);
     window.bus.$on(BUS_EVENTS.WEBSOCKET_DISCONNECT, () => {
-      this.updateOnlineStatus({ type: 'offline' });
+      // TODO: Remove this after completing the conversation list refetching
+      // TODO: DIRTY FIX : CLEAN UP THIS WITH PROPER FIX, DELAYING THE RECONNECT FOR NOW
+      // THE CABLE IS FIRING IS VERY COMMON AND THUS INTERFERING USER EXPERIENCE
+      setTimeout(() => {
+        this.updateOnlineStatus({ type: 'offline' });
+      }, 4000);
```

Comment:
@nithindavid, Can you check if it's working correctly? The set timeout is set to run 4 seconds after the WEBSOCKET_DISCONNECT event is detected, but the snack bar triggers immediately when I turn offline from the network tab. I think it's because of offline eventListener.
https://www.loom.com/share/0cffc02bfbb24360b74ac27a704c2f43?sid=3d3322d9-d9c2-4e94-986d-3a1f1a8193af

Reply:
Yes, we also have network state change. This 4 second delay is only for the action cable events.